plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/ResultSetModel.java

```diff
@@ -276,7 +276,8 @@ public void setAttributeVisibility(@NotNull DBDAttributeBinding attribute, boole
             constraint.setVisible(visible);
             if (attribute.getParentObject() == null || attribute.getParentObject() == documentAttribute) {
                 if (visible) {
-                    visibleAttributes.add(attribute);
+                    final int position = Math.min(constraint.getVisualPosition(), visibleAttributes.size());
+                    visibleAttributes.add(position, attribute);
```

Comment:
Looks like it will have IndexOutOfBounds for the second `visibleAttributes.size()`